src/wrapper.ts

```diff
@@ -227,28 +227,25 @@ export namespace Signal {
         const node = this[NODE];
         assertConsumerNode(node);
 
-        let indicesToShift = [];
-        for (let i = 0; i < node.producerNode.length; i++) {
+        for (let i = node.producerNode.length - 1; i >= 0; i--) {
           if (signals.includes(node.producerNode[i].wrapper)) {
             producerRemoveLiveConsumerAtIndex(node.producerNode[i], node.producerIndexOfThis[i]);
-            indicesToShift.push(i);
-          }
-        }
-        for (const idx of indicesToShift) {
-          // Logic copied from producerRemoveLiveConsumerAtIndex, but reversed
-          const lastIdx = node.producerNode!.length - 1;
-          node.producerNode![idx] = node.producerNode![lastIdx];
-          node.producerIndexOfThis[idx] = node.producerIndexOfThis[lastIdx];
-
-          node.producerNode.length--;
-          node.producerIndexOfThis.length--;
-          node.nextProducerIndex--;
-
-          if (idx < node.producerNode.length) {
-            const idxConsumer = node.producerIndexOfThis[idx];
-            const producer = node.producerNode[idx];
-            assertProducerNode(producer);
-            producer.liveConsumerIndexOfThis[idxConsumer] = idx;
+
+            // Logic copied from producerRemoveLiveConsumerAtIndex, but reversed
+            const lastIdx = node.producerNode!.length - 1;
+            node.producerNode![i] = node.producerNode![lastIdx];
+            node.producerIndexOfThis[i] = node.producerIndexOfThis[lastIdx];
```

Comment:
this stuff can be skipped if `lastIdx === i`

Reply:
You're correct, but I believe it won't make any difference. I prefer to implement only the necessary changes needed to fix the issue, making it easier for the owner to review.

Reply:
I certainly appreciate that! I also think it would make this code path slightly easier to read if the special case where the producer is the last item in the array was more clearly delineated like it is below the linked snippet.